python/src/scippneutron/_loading_detector_data.py

```diff
@@ -48,8 +49,24 @@ def _iso8601_to_datetime(iso8601: str) -> Optional[datetime]:
         return None
 
 
+def _convert_array_to_metres(array: np.ndarray, unit: str) -> np.ndarray:
+    if sc.Unit(unit) != sc.units.m:
+        return sc.to_unit(
+            sc.Variable(["temporary_variable"],
+                        values=array,
+                        unit=unit,
+                        dtype=np.float64), "m").values
+    return array
```

Comment:
I might suggest
```suggestion
        return sc.to_unit(
            sc.Variable(["temporary_variable"],
                        values=array,
                        unit=unit,
                        dtype=np.float64), "m").values
```
since with the rewrite for shallow copying we can refactor `to_unit` to not make a copy if the unit is unchanged.